subsys/bootloader/bl_validation/bl_validation.c

```diff
@@ -125,16 +131,34 @@ static bool validation_info_check(const struct fw_validation_info *vinfo)
 
 /* Find the validation_info at the end of the firmware. */
 static const struct fw_validation_info *
-validation_info_find(u32_t start_address, u32_t search_distance)
+validation_info_find(u32_t start_address, u32_t search_distance,
+		     const struct fw_info *finfo)
```

Comment:
As discussed OOB, the "offical" version of the combined firmware should be the version of the final image. This will probably require creating a dedicated looping function in fw_info, so the bootloader can also use it. When that is in place, the ext_api_provide function can also be updated to be aware of all the fw_infos in the chain.